### PR TITLE
Update Docs

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
## Purpose
$Subject

Fixes:https://github.com/ballerina-platform/ballerina-standard-library/issues/2569
## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

**Review by daneshk — APPROVED** [review]

File: ballerina/natives.bal [hunk 1]
```diff
@@ -21,7 +21,7 @@ import ballerina/jballerina.java;
 # boolean isMatched = regex:matches("Ballerina is great", "Ba[a-z ]+");
 # ```
 #
-# + stringToMatch - The string to match the regex
+# + stringToMatch - The string to match the `regex`
```
Comment (praneesha) [inline ballerina/natives.bal:24]:
```suggestion
# + stringToMatch - The string to match the `regex`
```
Why do we need backticks here?

File: ballerina/natives.bal [hunk 2]
```diff
@@ -37,7 +37,7 @@ public isolated function matches(string stringToMatch, string regex) returns boo
 #
 # + originalString - The original string to replace the occurrences of the
 #                    substrings that match the provided `regex`
-# + regex - The regex to match the substrings in the `originalString` to be replaced
+# + regex - The `regex` to match the substrings in the `originalString` to be replaced
```
Comment (praneesha) [inline ballerina/natives.bal:40]:
```suggestion
# + regex - The `regex` to match the substrings in the `originalString` , which is to be replaced
```
Why do we need backticks for the word 'regex' here?

File: ballerina/natives.bal [hunk 3]
```diff
@@ -62,7 +62,7 @@ public isolated function replaceAll(string originalString, string regex, string
 #
 # + originalString - The original string to replace the first occurrence of the
 #                    substring that matches the provided `regex`
-# + regex - The regex to match the first substring in the `originalString` to
+# + regex - The `regex` to match the first substring in the `originalString` to
```
Comment (praneesha) [inline ballerina/natives.bal:65]:
```suggestion
# + regex - The `regex` to match the first substring in the `originalString` to
```
Why do we need backticks for the word 'regex' here?

File: ballerina/natives.bal [hunk 4]
```diff
@@ -81,13 +81,13 @@ public isolated function replaceFirst(string originalString, string regex, strin
 }
 
 # Returns an array of strings by splitting a string using the provided
-# delimiter.
+# `regex` as the delimiter.
```
Comment (praneesha) [inline ballerina/natives.bal:84]:
```suggestion
# `regex` as the delimiter.
```
Why do we need backticks for the word 'regex' here?

File: ballerina/natives.bal [hunk 5]
```diff
@@ -81,13 +81,13 @@ public isolated function replaceFirst(string originalString, string regex, strin
...
 # ```ballerina
 # string[] result = regex:split("Ballerina is great", " ");
 # ```
 #
 # + receiver - The string to split
-# + delimiter - The delimiter to split by
+# + delimiter - The delimiter is a `regex` that splits the given string
```
Comment (praneesha) [inline ballerina/natives.bal:90]:
```suggestion
# + delimiter - The delimiter is a `regex`, which splits the given string
```
Why do we need backticks for the word 'regex' here?

File: docs/spec/spec.md [hunk 6]
```diff
@@ -22,7 +22,7 @@ makes it easier to use, combine, and create network services.
 
 # 1. Overview
 This library is based on [regular expressions](https://en.wikipedia.org/wiki/Regular_expression), which are notations 
-for describing sets of character strings that specify a search pattern.
+for describing sets of character strings that specify a search pattern. It supports the [regular expression pattern of Java](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/regex/Pattern.html#sum).
```
Comment (praneesha) [inline docs/spec/spec.md:25]:
```suggestion
for describing sets of character strings that specify a search pattern. It supports the [regular expression patterns of Java](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/regex/Pattern.html#sum).
```

File: ballerina/natives.bal (hunk 1 above)
Comment (kalaiyarasiganeshalingam) [reply ballerina/natives.bal:24]: @praneesha  Some have backticks but some haven't. So I added this to make it identical.
https://github.com/ballerina-platform/module-ballerina-regex/blob/main/ballerina/natives.bal#L32

Comment (kalaiyarasiganeshalingam) [reply ballerina/natives.bal]: As per the offline discussion, Remove all the backticks from the `regex` words as this is a normal word.

Comment (praneesha) [reply ballerina/natives.bal]: Ack. Thanks.

**Review by praneesha — APPROVED** [review]

File: docs/spec/spec.md [hunk 7]
```diff
@@ -3,7 +3,7 @@
 _Owners_: @daneshk @kalaiyarasiganeshalingam  
 _Reviewers_: @daneshk  
 _Created_: 2021/12/07   
-_Updated_: 2021/12/07  
+_Updated_: 2021/01/11
```
Comment (daneshk) [inline docs/spec/spec.md:6]:
```suggestion
_Updated_: 2022/01/11  
```

**Review by daneshk — APPROVED** [review]

**Pull request merged**